### PR TITLE
refactor miscellaneous daos and utilities

### PR DESCRIPTION
**Pull request by terrazoon**
## Description

Upgrade inbound_number_dao, inbound_sms_dao, complaint_dao, etc. to sqlalchemy 2.0

## Security Considerations

N/A

### CODE REVIEW

File: app/inbound_sms/rest.py [hunk 1]
```diff
@@ -60,9 +60,15 @@ def get_most_recent_inbound_sms_for_service(service_id):
     results = dao_get_paginated_most_recent_inbound_sms_by_user_number_for_service(
         service_id, int(page), limit_days
     )
-    return jsonify(
-        data=[row.serialize() for row in results.items], has_next=results.has_next
-    )
+    print(f"RESULTS ITEMS {results.items}")
+    print(f"RESULTS HAS_NEXT {results.has_next}")
```
Comment (xlorepdarkhelm) [inline app/inbound_sms/rest.py:64]: prints?

**Review by xlorepdarkhelm — CHANGES_REQUESTED** [review]
Looks good, but I saw a couple print statements, should they be removed?

File: app/inbound_sms/rest.py (hunk 1 above)
Comment (terrazoon) [reply app/inbound_sms/rest.py:64]: oops yes.  Thanks!

File: tests/app/dao/test_fact_notification_status_dao.py [hunk 2]
```diff
@@ -81,11 +81,11 @@ def test_fetch_notification_status_for_service_by_month(notify_db_session):
     )
 
     assert len(results) == 4
+    print(results)
```
Comment (ccostino) [inline tests/app/dao/test_fact_notification_status_dao.py:84]: Is this `print` still necessary?

**Review by ccostino — COMMENTED** [review]
Thanks, @terrazoon!  One quick question on another `print` statement but otherwise all is good!

**Review by ccostino — APPROVED** [review]
Thanks, @terrazoon! 🎉 

@xlorepdarkhelm, looks like you need to review again as well. 🙂

**Review by A-Shumway42 — APPROVED** [review]
I like the changes. Feels easier to read for me. No concerns here!

**Pull request merged**